src/server.tsx

```diff
@@ -65,59 +45,24 @@ export const createIncludeElement = (
   url.searchParams.append("sign", sign(url));
 
   esiAt.src = url.pathname + url.search;
-  let attrs = "";
-  Object.entries(esiAt).forEach(
-    ([key, value]) => (attrs += ` ${key}="${value ? escapeAttr(value) : ""}"`)
-  );
 
-  return `<esi:include${attrs} />`;
+  return React.createElement("esi:include", esiAt);
 };
 
-/**
- * Removes the placeholder holding the data-reactroot attribute.
- */
-class RemoveReactRoot extends Transform {
-  public skipStartOfDiv = true;
-  public bufferedEndOfDiv = false;
-
-  // eslint-disable-next-line @typescript-eslint/no-explicit-any
-  public _transform(chunk: any, encoding: string, callback: TransformCallback) {
-    // '<div data-reactroot="">'.length is 23
-    chunk = chunk.toString();
-    if (this.skipStartOfDiv) {
-      // Skip the wrapper start tag
-      chunk = chunk.substring(23);
-      this.skipStartOfDiv = false;
-    }
-
-    if (this.bufferedEndOfDiv) {
-      // The buffered end tag wasn't the last one, push it
-      chunk = "</div>" + chunk;
-      this.bufferedEndOfDiv = false;
-    }
-
-    if (chunk.substring(chunk.length - 6) === "</div>") {
-      chunk = chunk.substring(0, chunk.length - 6);
-      this.bufferedEndOfDiv = true;
-    }
-
-    callback(undefined, chunk);
-  }
-}
-
 interface IServeFragmentOptions {
```

Comment:
Is this interface still necessary now that we don't need to modify the outputted HTML thanks to your trick with `React.createElement()`?

Reply:
It depends, it looks like that pipestream was there so that users could customize the stream, not necessarily for removing the surrounding div from the esi tag ( this was handled independently from the pipeStream option

Reply:
ok!